src/iocp/mod.rs

```diff
@@ -698,9 +783,14 @@ impl PacketUnwrapped {
                 socket.mode = mode;
                 socket.interest_error = true;
 
+                // If there was a change, indicate that we need an update.
                 match socket.status {
-                    SocketStatus::Polling { readable, writable } => {
-                        (interest.readable && !readable) || (interest.writable && !writable)
+                    SocketStatus::Polling { flags } => {
+                        let our_flags =
+                            event_to_afd_mask(interest.readable, interest.writable, true)
```

Comment:
why is this given `true` as the last argument?

Reply:
Because `interest_error` in this case is always `true`. I've clarified it here.